File: 30days/11/11.rb
#!/usr/bin/env ruby

arr = Array.new(6)
(0...6).each do |arr_i|
    arr_t = gets.strip
    arr[arr_i] = arr_t.split(" ").map(&:to_i)
end

total = 0
max = false

(0..3).each do |i|
    (0..3).each do |j|
        total = arr[i][j] + arr[i][j + 1] + arr[i][j + 2]
        total += arr[i + 1][j + 1]
        total += arr[i + 2][j] + arr[i + 2][j + 1] + arr[i + 2][j + 2]
        max = total if max == false || total > max
    end
end

puts max

File: 30days/11/11.py
#!/usr/bin/env python

arr = []
for arr_i in range(6):
    arr_t = [int(arr_temp) for arr_temp in input().strip().split(' ')]
    arr.append(arr_t)

total = 0
maxim = False

for i in range(0, 4):
    for j in range(0, 4):
        total = arr[i][j] + arr[i][j + 1] + arr[i][j + 2]
        total += arr[i + 1][j + 1]
        total += arr[i + 2][j] + arr[i + 2][j + 1] + arr[i + 2][j + 2]
        if maxim is False or total > maxim:
            maxim = total

print(maxim)
